src/common/src/catalog/mod.rs

```diff
@@ -528,6 +529,37 @@ impl ConflictBehavior {
     }
 }
 
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum Engine {
+    /// TODO(nimtable): use iceberg engine as default.
+    #[default]
+    Hummock,
+    Iceberg,
+}
+
+impl Engine {
```

Comment:
You can implement from trait for Engine directly. (from PbEngine, from &PbEngine)